Support multiple monitors through Xinerama

d3adb5 opened the pull request:
This adds support for multiple monitors through Xinerama if present at the time of compiling. The X and Y coordinates now take into consideration the target monitor for stalonetray's window (defaults to 0).

These changes also remove Cirrus CI config and add a CI workflow for GitHub Actions, which might not be in the upstream project's interest currently. Let me know if you'd like me to cherry-pick Xinerama changes to another branch and submit another PR, @kolbusa.

FYI I'm more than willing to take over maintainership of this project if you're interested in transferring it. Let me know, :eyes: 

Closes #37.

d3adb5 committed:
ci: switch from cirrus-ci to github actions

Switch from Cirrus CI to GitHub Actions for continuous integration.

d3adb5 committed:
refactor: adjust format strings to eliminate warnings

Adjust format strings to follow compiler recommendations and eliminate
warnings.

d3adb5 committed:
ci: rename ci job for proper capitalization

Add a name to the one CI job in the CI workflow to use proper
capitalization.

d3adb5 committed:
Merge pull request #1 from d3adb5/refactor/eliminate-warnings

refactor: adjust format strings to eliminate warnings

d3adb5 committed:
fix: skip manpage generation if unsupported

Skip generating manpage recipes if the system lacks xsltproc.

d3adb5 committed:
feat: support multiple monitors through xinerama

Support multiple monitors using the Xinerama extension. This introduces
a new flag / setting to select the target monitor for the tray window.
The geometry setting is now relative to that monitor.

d3adb5 committed:
ci: install libxinerama-dev for xinerama support

Install the Xinerama extension to enable Xiinerama during CI and
release.

d3adb5 committed:
docs: document new --monitor flag

Document the new --monitor flag and configuration option.

d3adb5 committed:
Merge pull request #2 from d3adb5/feat/support-xinerama

feat: support multiple monitors through xinerama

d3adb5 committed:
docs: update readme with mentions of xinerama

Update project README.md to include information about Xinerama.

d3adb5 committed:
Merge pull request #3 from d3adb5/docs/update-readme

docs: update readme with mentions of xinerama

d3adb5 committed:
ci: cache dependencies to speed up builds

Cache dependencies in GitHub Actions to speed up build times for CI.

d3adb5 committed:
Merge pull request #4 from d3adb5/ci/cache-dependencies

ci: cache dependencies to speed up builds

kolbusa reviewed (commented):
Hi @d3adb5 ! Thanks for the PR and for the offer to take over the maintenance -- this is much appreciated and I'll figure out the next steps after this PR is merged. I left one comment regarding the CI, but I need more time to review the rest.

d3adb5 committed:
ci: run builds on freebsd 14.2 vm

Run builds on FreeBSD 14.2 parallel to Ubuntu builds.

d3adb5 committed:
ci: cache virtualbox dependencies

Cache VirtualBox dependencies to speed up FreeBSD CI runs.

d3adb5 committed:
ci: execute install scripts for vbox deps

Execute install scripts for VirtualBox dependencies to try skipping
vbox.sh's apt-get installs.

d3adb5 committed:
ci: disable man-db auto update to speed up installs

Disable man-db auto updates to speed up package installation on Ubuntu.

d3adb5 committed:
Merge pull request #5 from d3adb5/ci/run-on-freebsd

ci: run builds on freebsd 14.2 vm

d3adb5 commented on .cirrus.yml:
```diff
@@ -1,41 +0,0 @@
-    make_script:
-        - make
-
-build_freebsd_x86_64_task:
```
Not _natively_, but I just added a job to the CI workflow that uses [vmactions/freebsd-vm](https://github.com/vmactions/freebsd-vm) to compile within a FreeBSD 14.2 VM. Let me know what you think, @kolbusa.

kolbusa replied:
LGTM, thanks!

kolbusa commented:
Thanks for the PR! Looks good to me. Merging!
